### ISSUE TITLE
[Bug]: deleting from blank tab MRU gets URLs and site names out of sync

### ISSUE THREAD

**Issue by JustinWebDev** [mozilla-mobile/fenix#20431]
### Steps to reproduce

1. have some MRU of more than (8) to show on a new tab
2. delete one of the MRU items via the thumbnail or an close via an open page
- observe now the site favicons don't match the site names

In this pic you can easily see lowes.com, Google, and VUit don't match.
![image](https://user-images.githubusercontent.com/40804543/126241281-74c26cbc-6f37-4a86-bba1-75e5199ebc44.png)

### Expected behaviour

favicons and names should not be out of sync

### Actual behaviour

They do get out of sync.  In the pic you can easily see lowes.com, Google, and VUit don't match, even though all but one don't match.

### Device name

Samsung Galaxy Tab A SM-T280

### Android version

5.1.1

### Firefox release type

Firefox

### Firefox version

90.1.1

**Comment by SoftVision-LorandJanos** [comment]
Hi @JustinWebDev ! Thanks for your report!
I can reproduce the issue you described, with a little difference:
 - On Google Pixel 4 (Android 11) I had a full icon for a MRU and for the same page one with only an initial on it.
 - On Samsung Galaxy Tab 3 (Android 9) I had the result without deleting anything, and without filling all 8 places.
 - On LG Nexus 5 (Android 6.0.1) I saw a switch between the icons for certain pages.
 As I noticed it happens randomly after a few pages are accessed.
